FirebaseDatabase/Tests/Integration/FIRDatabaseQueryTests.m

```diff
@@ -2946,6 +3554,23 @@ - (void)testChildChangedCausesChildRemovedEvent {
   WAIT_FOR(removed);
 }
 
+- (void)testChildChangedCausesChildRemovedEventWithStartAfter {
+  FIRDatabaseReference* ref = [FTestHelpers getRandomNode];
+  [[ref child:@"l/b"] setValue:@"2" andPriority:@"b"];
+  FIRDatabaseQuery* query = [[[[ref child:@"l"] queryOrderedByPriority]
+      queryStartingAfterValue:@"a"] queryEndingAtValue:@"d"];
+  __block BOOL removed = NO;
+  [query observeEventType:FIRDataEventTypeChildRemoved
+                withBlock:^(FIRDataSnapshot* snapshot) {
+                  XCTAssertEqualObjects(snapshot.value, @"2", @"Incorrect snapshot");
+                  removed = YES;
+                }];
+
+  [[ref child:@"l/b"] setValue:@"4" andPriority:@"a"];
```

Comment:
Can you explain why this matches the `startAfter:a` query?

Reply:
It's because `query` is on child `l`, not `l/b`. But now that I'm looking into this I'm a bit confused about the behavior here. I don't think the `child_removed` event is supposed to fire at all if it's not an "immediate child": 

https://firebase.google.com/docs/database/admin/retrieve-data#child-removed

Here, we are updating `@"l/b"`, which is an immediate child of `@"l"`, but it's not removed. If anything, I'd expect `child_change` to fire:

https://firebase.google.com/docs/database/admin/retrieve-data#child-changed

Reply:
Kinda strange, I added an equivalent test to the android SDK:

https://github.com/firebase/firebase-android-sdk/commit/80170e8aaf2b063bf6b3223ef9ec727fb1ab8921

And in that case the `child_removed` event doesn't fire. In any case, I think I intended this test here to be similar to the one above (including @"l/a"). Since I don't want to block this PR on figuring out why this discrepancy exists, I'm going to push a the intended version of the test and circle back to this case later.

Reply:
Hm. I think I understand the test now. We add a value with priority b, which matches the startAfter. Then we change the priority to "a" and the node no longer matches. Initially, I thought the second write added another node (since the value changed), but it just updates the value and priority. I'll let you decide if I am right :)